lib/bloc/podcast_clip/podcast_clip_bloc.dart

```diff
@@ -0,0 +1,238 @@
+import 'package:path_provider/path_provider.dart';
+import 'package:rxdart/rxdart.dart';
+
+enum PodcastClipState {
+  IDLE,
+  FETCHING_IMAGE,
+  FETCHING_AUDIO_CLIP,
+  GENERATING_CLIP
+}
+
+const int _initialSeconds = 10;
+const int _maxClipDuration = 120;
+
+class PodcastClipBloc with AsyncActionsHandler {
+  final BehaviorSubject<PodcastClipDetailsModel> _clipDetailsBehaviourSubject =
+      BehaviorSubject();
+
+  Stream<PodcastClipDetailsModel> get clipDetails =>
+      _clipDetailsBehaviourSubject.stream;
+
+  final BehaviorSubject<PodcastClipState> _clipStateBehaviourSubject =
+      BehaviorSubject();
+
+  Stream<PodcastClipState> get clipState => _clipStateBehaviourSubject.stream;
+
+  setPodcastClipDetails({PositionState position}) {
+    _clipStateBehaviourSubject.add(PodcastClipState.IDLE);
+    Duration endTime = position.position + Duration(seconds: _initialSeconds);
+
+    PodcastClipDetailsModel podcastClipDetails = PodcastClipDetailsModel(
+        startTimeStamp: position.position,
+        endTimeStamp: endTime,
+        clipDuration: _initialSeconds,
+        episodeLength: position.length,
+        episodeDetails: position.episode);
+
+    _clipDetailsBehaviourSubject.add(podcastClipDetails);
+  }
+
+  setPodcastClipDuration({int durationInSeconds}) async {
+    var clipDetails = await getCurrentPodcastClipDetails();
+
+    if ((clipDetails.endTimeStamp + Duration(seconds: durationInSeconds)) >
+        clipDetails.episodeLength) {
+      return;
+    }
+
+    Duration clipEndTime = clipDetails.endTimeStamp +
+        Duration(seconds: durationInSeconds) -
+        Duration(seconds: clipDetails.clipDuration);
+
+    _clipDetailsBehaviourSubject.add(clipDetails.copy(
+        clipDuration: durationInSeconds, endTimeStamp: clipEndTime));
+  }
+
+  incrementDuration() async {
+    var clipDetails = await getCurrentPodcastClipDetails();
+    int currentClipDuration = clipDetails.clipDuration;
+
+//If  duration is less than 10 seconds return
+    if (_maxClipDuration - currentClipDuration <= 0) {
+      return;
+    }
+    if ((clipDetails.endTimeStamp + Duration(seconds: _initialSeconds)) >
+        clipDetails.episodeLength) {
+      return;
+    }
+    int incrementedClipDuration = 0;
+    Duration incrementedClipEndTime = Duration(seconds: 0);
+    if (currentClipDuration % 10 != 0) {
+      incrementedClipDuration =
+          currentClipDuration - currentClipDuration % 10 + _initialSeconds;
+      incrementedClipEndTime = Duration(
+          seconds:
+              clipDetails.startTimeStamp.inSeconds + incrementedClipDuration);
+    } else {
+      incrementedClipDuration = currentClipDuration + _initialSeconds;
+      incrementedClipEndTime = Duration(
+          seconds: clipDetails.endTimeStamp.inSeconds + _initialSeconds);
+    }
+
+    _clipDetailsBehaviourSubject.add(clipDetails.copy(
+        clipDuration: incrementedClipDuration,
+        endTimeStamp: incrementedClipEndTime));
+  }
+
+  decrementCount() async {
```

Comment:
decrementDuration maybe?
Same as with incrementDuration if possible to reuse the `setPodcastClipDuration` it would leave a single validation point.

Reply:
Done. Thanks for highlighting it.